src/test/java/com/aws/greengrass/device/session/AbstractSessionFactoryTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.device.session;
+
+import com.aws.greengrass.device.exception.AuthenticationException;
+import com.aws.greengrass.testcommons.testutilities.GGExtension;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.HashMap;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@ExtendWith({MockitoExtension.class, GGExtension.class})
+public class AbstractSessionFactoryTest {
+    private static final String mqttCredentialType = "mqtt";
+    private static final String unknownCredentialType = "unknown";
+
+    @Mock
+    private MqttSessionFactory mqttSessionFactory;
+
+    @AfterEach
+    void afterEach() {
+        AbstractSessionFactory.unregisterSessionFactory(mqttCredentialType);
+    }
+
+    @Test
+    void GIVEN_noRegisteredFactories_WHEN_createSession_THEN_throwsException() {
+        Assertions.assertThrows(IllegalArgumentException.class,
```

Comment:
Illegal argument is unchecked, so callers will need to be careful to handle it fully. Our coding guidelines also say to prefer checked exceptions where possible, so consider that as well.